CONTRIBUTING.md

```diff
@@ -1,71 +1,73 @@
 
-1. Prior to investing a large amount of time into significant or invasive changes, or those that affect the output, it is likely more efficient to first open an issue for discussion and planning.
-2. If you are not familiar with the project, focus first on the [Good first issues](https://github.com/ocaml-ppx/ocamlformat/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22Good-first-issue+%3A%2B1%3A%22).
+1. For significant, invasive, or output-affecting changes, consider opening an issue for discussion before committing substantial time and effort.
+2. If you're new to the project, starting with the [Good first issues](https://github.com/ocaml-ppx/ocamlformat/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22Good-first-issue+%3A%2B1%3A%22) can be beneficial.
 3. Fork the repository and create your branch from `main`.
-4. If you have added code that should be tested, add tests (they should be located in the `tests/passing` directory).
-'.
-5. Ensure the test suite passes (see [Running the tests](#running-the-tests)).
-
-When acknowledged, the project maintainers will add [labels](#ocamlformat-labels) to your pull request throughout its lifespan.
+4. If you've added code that should be tested, supplement it with tests located in the `tests/passing` directory.
+5. Ensure that the test suite passes (see [Running the tests](#running-the-tests) for instructions).
 
+After acknowledgment, maintainers will tag your pull request with appropriate [labels](#ocamlformat-labels).
 
-### Running the tests
+### Running the Tests
 
-Once OCamlFormat has been built, run `make test` to check for regressions.
+After building OCamlFormat, execute `make test` to check for any regressions. Here's what happens during `make test`:
 
-The first step of `make test` is to ensure that invoking OCamlFormat on its own source code produces the same source code (you can run `make fixpoint` to only check this). If OCamlFormat is not integrated with the editor you use, you should run `make fmt` to reformat all OCamlFormat source files.
+1. The command checks that running OCamlFormat on its own source code yields an identical result. Run `make fixpoint` for this check only. If OCamlFormat is not integrated with your editor, use `make fmt` to reformat all OCamlFormat source files.
+2. The command ensures the test suite passes. No unexpected diffs should appear. If any, accept all diffs at once with `make regtests-promote`.
 
-The second part of `make test` is to ensure the test suite passes. You should not see any unexpected diff. One can accept all diffs at once with `make regtests-promote`.
+Once `make test` passes, test your pull requests on code in external repositories by executing `tools/test_branch.sh <rev>`, where `<rev>` is the git revision/branch with your changes. If your pull request modifies an option, also run `OCAMLFORMAT=<option>=<value> tools/test_branch.sh <rev>` to test with the option enabled.
```

Comment:
This script does not take a `<rev>` argument anymore. It takes revisions as flags but detects the right revisions by default.

The ocamlformat options are now passed as arguments: `tools/test_branch.sh "<option1>=<value>,<option2>=<value>"`.